tests/integ/scala/test_object_already_exists.py

```diff
@@ -0,0 +1,74 @@
+#
+# Copyright (c) 2012-2025 Snowflake Computing Inc. All rights reserved.
+#
+
+import pytest
+from snowflake.snowpark._internal.utils import set_ast_state, AstFlagSource
+from snowflake.snowpark.exceptions import SnowparkSQLException
+
+pytestmark = [
+    pytest.mark.xfail(
+        "config.getoption('local_testing_mode', default=False)",
+        reason="This is a SQL test suite",
+        run=False,
+    ),
+]
+
+
+@pytest.fixture(autouse=True)
+def setup(request, session):
+    """Setup fixture to enable AST for all tests in this module."""
+    original = session.ast_enabled
+    set_ast_state(AstFlagSource.TEST, True)
+    yield
+    set_ast_state(AstFlagSource.TEST, original)
+
+
+def test_existing_table_with_save_as_table(session):
+    table_name = "DUPLICATE_TABLE"
+    df = session.create_dataframe([{"a": 1, "b": 2}])
```

Comment:
generate table name using random table name util to avoid conflits and flakiness here and other tests.